measures/ResStockArgumentsPostHPXML/measure.rb

```diff
@@ -100,71 +100,67 @@ def run(model, runner, user_arguments)
     max_random_shift_steps = (args[:flex_random_shift_minutes] / @minutes_per_step).to_i
     @random_shift_steps = @prng.rand(-max_random_shift_steps..max_random_shift_steps)
 
-    # Parse the HPXML document
-    doc = XMLHelper.parse_file(@hpxml_path)
-    hpxml_doc = XMLHelper.get_element(doc, '/HPXML')
-    doc_buildings = XMLHelper.get_elements(hpxml_doc, 'Building')
-
     # Process each building
-    doc_buildings.each_with_index do |building, index|
-      hpxml_bldg = @hpxml.buildings[index]
+    @hpxml.buildings.each_with_index do |hpxml_bldg, index|
       if hpxml_bldg.hvac_controls.to_a.length != 0 && !skip_hvac_flexibility?(args)
         hvac_schedule = create_hvac_schedule(index)
-        modified_schedule = modify_hvac_schedule(index, hvac_schedule)
-        write_schedule(modified_schedule, building, index, output_csv_path)
-      end
-      if !skip_ev_flexibility?(args)
-        ev_schedule = get_ev_schedule(building)
-        next if ev_schedule.nil?
-        modified_ev_schedule = modify_ev_schedule(index, ev_schedule)
-        write_schedule(modified_ev_schedule, building, index, output_csv_path)
+        modified_schedule = modify_hvac_schedule(hpxml_bldg, hvac_schedule)
+        write_schedule(modified_schedule, hpxml_bldg, index, output_csv_path)
       end
+      next unless !skip_ev_flexibility?(args)
+
+      ev_schedule = get_ev_schedule(hpxml_bldg)
+      next if ev_schedule.nil?
+
+      modified_ev_schedule = modify_ev_schedule(hpxml_bldg, ev_schedule)
+      write_schedule(modified_ev_schedule, hpxml_bldg, index, output_csv_path)
     end
 
     # Write out the modified hpxml
-    XMLHelper.write_file(doc, @hpxml_path)
+    XMLHelper.write_file(@hpxml.to_doc(), @hpxml_path)
     runner.registerInfo("Wrote file: #{@hpxml_path} with modified schedules.")
-    true
+    return true
```

Comment:
I think that rubocop by default removes the `return` syntax, but I think that is confusing (especially to developers newer to ruby) so I made them all explicit.